templates/monitoring/versions.tf.tpl

```diff
@@ -5,7 +5,7 @@ terraform {
   required_providers {
     datadog = {
       source  = "datadog/datadog"
-      version = ">= 2.20.0"
+      version = ">= 3.0.0, < 3.27.0"
```

Comment:
Just clarifying what Nate said: We saw issues with a specific version. If `3.27.0` works fine (if Datadog fixed their provider issue), then you should just pin to a major version range:

```suggestion
      version = ">= 3.0.0, < 4.0.0"
```
If you didn't confirm it worked on this version, you should pin to the latest that works.

Reply:
Looks like they did fix that issue, so updating to use anything between 3 and 4 (exclusive)